File: PlotPocket.Client/src/app/models/show.dto.ts
export enum ShowType {
  Movie = 'Movie',
  TvShow = 'TvShow',
}

export interface ShowDto {
  id: number;
  title: string;
  overview: string;
  PosterPath?: string;
  posterPath?: string;
  rating: number;
  ReleaseDate?: Date;
  type: ShowType;
  isWatchlisted: boolean;
  isInWatchlist: boolean;
  isWatched: boolean;
  isBookmarked: boolean;
}

File: PlotPocket.Server/Models/Dtos/ShowDto.cs
using System.Text.Json.Serialization;

namespace PlotPocket.Server.Models.Dtos;

public class ShowDto
{
    public int Id { get; set; }
    public string Title { get; set; }
    public string Overview { get; set; }
    public string PosterPath { get; set; }
    public double Rating { get; set; }
    public DateTime? ReleaseDate { get; set; }

    [JsonConverter(typeof(JsonStringEnumConverter))]
    public ShowType Type { get; set; }

    public bool IsWatchlisted { get; set; }
    public bool IsWatched { get; set; }
    public bool IsBookmarked { get; set; }
}

public enum ShowType
{
    Movie,
    TvShow
}
